cli/src/pcluster/config/cluster_config.py

```diff
@@ -65,49 +70,279 @@ def __repr__(self):
         )
 
 
-class ImageConfig(Config):
-    """Represent the configuration of an Image."""
+# ---------------------- Storage ---------------------- #
+class BaseEbsConfig(Config):
+    """Represent the configuration shared by EBS and RootVolume."""
 
-    def __init__(self, os: str, id: str = None):
+    def __init__(
+        self,
+        volume_type: str = None,
+        iops: int = None,
+        size: int = None,
+        encrypted: bool = None,
+        kms_key_id: str = None,
+        throughput: int = None,
+    ):
         super().__init__()
-        self.os = os
+        if volume_type is None:
```

Comment:
For the simplest conditions you can have a single row for the initialization:
```python
self.volume_type = MarkedStr("gp2") if volume_type is None else volume_type
```

In any case I'd keep the conditions and the initialization of the attribute grouped together.

Reply:
Yes. I took the current approach to group the defaults together. With the simplest conditions, things will mix together.
Here I am wasting some lines of code for better readability.